Update AFSQLManager.m

shortstuffsushi opened the pull request:
Fixed issue when no error, but also not open. Completion block would never be called.
Updated block in createDatabase to just pass status and error back to completion block.

shortstuffsushi committed:
Update AFSQLManager.m

Fixed issue when no error, but also not open. Completion block would never be called.
Updated block in createDatabase to just pass status and error back to completion block.

AlvaroFranco commented:
I can't see your point. My code handles the error and the bool correctly. Moreover, your code doesn't handle when the user creates the database correctly but doesn't want to open it inmediately. With your commit, that would make the completion block return a failed status, when the database is created correctly. Please, check it out carefuly.

shortstuffsushi committed:
Update AFSQLManager.m

Misunderstood 'open' parameter. However, in the case of non-openImmediately requests, you still need to call that 'completion' block.
Also, fixed a typo in the method declaration... will probably need to find uses and fix them.

shortstuffsushi commented:
Closing this issue since it's only touching one file, and instead needs to affect all references (with that typo). Apparently my fork doesn't contain these changes I've made, so I'm just going to rewrite them all there.

shortstuffsushi commented:
Actually, it apparently made a branch, so I'll just use that.

shortstuffsushi committed:
Fixing typos in other files

AlvaroFranco commented:
Yeah, that's true, I wasn't handling that. Also, I guess the property could have a more-descriptive name. Merging, thanks!

shortstuffsushi commented:
I had just skimmed the method, and saw the open variable, and didn't realize it was being past in -- I thought it was returned by one of the functions you were calling. The flow makes sense, after actually reading through it, my bad.
